### PR TITLE
[release/6.0] Switch to dynamic cert gen for tests

### PR DESCRIPTION
**Pull request by halter73**
A lot of our test certs in release/6.0's `src/Shared/test/Certificates/` directory expired on Friday, May 17th. This PR cherry-picks #39685 to dynamically generate these certs so they don't expire again. This should unblock our dependency update at #55798.

### CODE REVIEW

File: src/Security/Authentication/test/CertificateTests.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 namespace Microsoft.AspNetCore.Authentication.Certificate.Test
 {
-    public class ClientCertificateAuthenticationTests
+    public class ClientCertificateAuthenticationTests : LoggedTest
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/CertificateTests.cs:27]: ❤️

**Review by wtgodbe — APPROVED** [review]

**Comment by wtgodbe** [comment]
Marking as `tell-mode` since this is test-only

**Comment by halter73** [comment]
@dotnet/aspnet-build Can someone merge this?

**Comment by BrennanConroy** [comment]
> Can someone merge this?

Patch merge windows apply to tell-mode changes too. You'll need to wait for the next window.

**Pull request merged**